spring-cloud-gcp-pubsub-stream-binder/src/test/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisionerTests.java

```diff
@@ -141,49 +141,10 @@ void testProvisionConsumerDestination_customSubscription() {
     assertThat(result.getName()).isEqualTo("my-custom-subscription");
   }
 
-  @Test
-  void testProvisionConsumerDestination_noTopicException() {
-    when(this.pubSubConsumerProperties.isAutoCreateResources()).thenReturn(false);
-    when(this.pubSubAdminMock.getTopic("topic_A")).thenReturn(null);
-
-    assertThatExceptionOfType(ProvisioningException.class)
-        .isThrownBy(
-            () ->
-                this.pubSubChannelProvisioner.provisionConsumerDestination(
-                    "topic_A", "group_A", this.properties))
-        .withMessage("Non-existing 'topic_A' topic.");
-  }
-
-  @Test
-  void testProvisionConsumerDestination_noSubscriptionException() {
-    when(this.pubSubConsumerProperties.isAutoCreateResources()).thenReturn(false);
-
-    assertThatExceptionOfType(ProvisioningException.class)
-        .isThrownBy(
-            () ->
-                this.pubSubChannelProvisioner.provisionConsumerDestination(
-                    "topic_A", "group_A", this.properties))
-        .withMessage("Non-existing 'topic_A.group_A' subscription.");
-  }
-
-  @Test
-  void testProvisionConsumerDestination_wrongTopicException() {
-    when(this.pubSubConsumerProperties.isAutoCreateResources()).thenReturn(false);
-    when(this.pubSubAdminMock.getSubscription("topic_A.group_A"))
-        .thenReturn(Subscription.newBuilder().setTopic("topic_B").build());
-
-    assertThatExceptionOfType(ProvisioningException.class)
-        .isThrownBy(
-            () ->
-                this.pubSubChannelProvisioner.provisionConsumerDestination(
-                    "topic_A", "group_A", this.properties))
-        .withMessage("Existing 'topic_A.group_A' subscription is for a different topic 'topic_B'.");
-  }
```

Comment:
Previously:
> When auto-create-resources is false, and `topicA.groupName` corresponds to a subscription assigned to `topicB`, then throw a ProvisioningException.

Now:
> When auto-create-resources is false, and `topicName.groupName` is provided, then do not check for the existence of the subscription or check its topic assignment. Instead, return a new PubSubConsumerDestination instance.

Like the above, this seems like an unintentional change in behavior outside the scope of the primary ensureTopicExists change. If it's intended, please provide some clarification.

Reply:
As behaviour is changed, subscription will be verified only if `isAutoCreateResources` is set to true. So I have added a new test instead with a different name `testProvisionConsumerDestination_subscriptionHasDifferentTopic`. I can rename it back to previous if required